packages/core/src/collision/CollisionDetection.ts

```diff
@@ -55,43 +47,50 @@ export class CollisionDetection extends Script {
         for (let i = 0, len = colliders.length; i < len; i++) {
           const collider = colliders[i];
           if (collider != this._myCollider && this._boxCollision(collider)) {
-            overlopCollider = collider;
-            // @ts-ignore
-            this.dispatch("collision", { collider });
+            overlappedCollider = collider;
+            let scripts = this.entity._scripts;
+            scripts._elements.forEach((script) => {
```

Comment:
forEach is slow, use `for(let i=0;i<n;i++)` is better, you can do a performacne test.  This fun is call eveny frame,  so performance is very important.